packages/babel-plugin-react-server/src/index.js

```diff
@@ -1,35 +1,82 @@
 import loggerSpec from 'react-server-module-tagger';
 import path from 'path';
+import * as t from "babel-types";
 
 module.exports = function() {
 	return {
 		visitor: {
 			Identifier(p, state) {
 				const {node} = p;
 				const {name} = node;
+				const tokens = getTokens(state);
 
-				const trim = state.opts.trim;
-				const prefix = state.opts.prefix;
-				const parent = path.resolve(path.join(process.cwd(), '..')) + path.sep;
-				const filePath = this.file.opts.filename.replace(parent, '');
-
-				//TODO: Support labels
-				const moduleTag = loggerSpec({ filePath, trim, prefix });
+				if (tokens.has(name)) {
+					const trim = state.opts.trim;
+					const prefix = state.opts.prefix;
+					const parent = path.resolve(path.join(process.cwd(), '..')) + path.sep;
+					const filePath = this.file.opts.filename.replace(parent, '');
+					const options = JSON.parse(loggerSpec({ filePath, trim, prefix }));
 
-				let tokens;
-				if (state.opts.tokens) {
-					tokens = new Set(state.opts.tokens);
-				} else {
-					tokens = new Set(["__LOGGER__", "__CHANNEL__", "__CACHE__"]);
+					p.replaceWith(convertObjectToObjectExpression(options));
 				}
+			},
+			CallExpression(p, state) {
+				const {node} = p;
+				const {callee} = node;
+				const {name} = callee;
+				const tokens = getTokens(state);
 
 				if (tokens.has(name)) {
-					// this strikes me as a dirty, nasty hack.  I think it would be better
-					// to parse the object as json and coerce it to an array of
-					// ObjectProperties to construct an ObjectExpression
-					p.node.name = moduleTag;
+					const trim = state.opts.trim;
+					const prefix = state.opts.prefix;
+					const parent = path.resolve(path.join(process.cwd(), '..')) + path.sep;
+					const filePath = this.file.opts.filename.replace(parent, '');
+					const options = JSON.parse(loggerSpec({ filePath, trim, prefix }));
+
+					// We're assuming that the options string will only ever contain a
+					// single option, `label`, and that users won't call a tag as a
+					// function without providing a label.  It's the only key we support
+					// (at least, its the only one documented in the gulp module tagger),
+					// but if we ever support more, we'll need to pass the arguments as a
+					// string to loggerSpec (probably by using babel-generator to generate
+					// the code again from the ast) or change the module tagger to take an
+					// options object, rather than a string.
+					const arg = node.arguments[0];
+					options.label = arg.properties[0].value.value;
```

Comment:
`loggerSpec` actually expects the label to be passed in, and it incorporates it into the `name`.

So I think something like this would work:

```javascript
const label = arg.properties[0].value.value;
const options = JSON.parse(loggerSpec({ filePath, trim, prefix, label }));
```

Though we might be able to more confidently pull out the value associated with a `label` key (rather than the value associated with the _first_ key) with something like:

```javascript
const {label} = arg.properties.reduce((o,p) => (o[p.key.value] = p.value.value, o), {});
```

Not familiar with babel plugins, so don't know if that's the right structure for the entries in `arg.properties`.